tests/cases/compiler/controlFlowFavorAssertedTypeThroughTypePredicate.ts

```diff
@@ -0,0 +1,22 @@
+// @noEmit: true
+// @strict: true
+
+// repro 49988#issuecomment-1192016929
+
+declare function isObject(value: unknown): value is Record<string, unknown>;
+
+declare const obj1: {};
+obj1;  // {}
+if (isObject(obj1)) {
+    obj1;  // Record<string, unknown>
+    obj1['attr'];
+}
+obj1;  // Record<string, unknown>
+
+declare const obj2: {} | undefined;
+obj2;  // {} | undefined
+if (isObject(obj2)) {
+    obj2;  // Record<string, unknown>
+    obj2['attr'];
+}
+obj2;  // Record<string, unknown> | undefined
```

Comment:
I know it's unfair to ask this of you since there _were no tests_, but:

* I would remove the comments - the type baselines and errors will indicate these better and not need to be updated.
* I would also add test cases for a type predicate like `value is {}` acting on `Record<string, unknown>` and `Record<string, unknown> | undefined`.

Reply:
No problem - I've added the requested test cases and removed those comments (I've originally kept them as they are pretty popular in your test suite)

Reply:
> I've originally kept them as they are pretty popular in your test suite

Yeah, I am not a fan 😅